src/repositories/SearchRepository.ts

```diff
@@ -0,0 +1,41 @@
+import { collection, query, where, getDocs } from 'firebase/firestore'
+import { database } from './Firebase'
+
+export class SearchRepository {
+  public async search(keyword: any) {
+    const userQuery = query(
+      collection(database, 'user'),
+      where('name', '>=', keyword),
+      where('name', '<=', keyword + '\uf8ff')
+    )
+    const roleQuery = query(
+      collection(database, 'role'),
+      where('name', '>=', keyword),
+      where('name', '<=', keyword + '\uf8ff')
```

Comment:
symbol greater than equal dan symbol less than equal itu fungsinya buat apa ya?

Reply:
it's an alternative way to use 'like' methods
>= name is have another character more than keyword
<= name can be less than keyword with maximum range of unicode

you can try it on firebase using menu query builder